File: July/31 July/department-service/src/main/java/com/example/model/Department.java
package com.example.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name="department_table")
public class Department {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int departmentId;
	private String departmentName;
	private String location;
	
	public Department(String departmentName, String location) {
		super();
		this.departmentName = departmentName;
		this.location = location;
	}
}

File: Aug/2 Aug/TypeScriptExamples/Department.ts
export class Department
{
    constructor(private _departmentId:number, private _departmentName:string) {
    }


    get departmentId(): number {
        return this._departmentId;
    }

    set departmentId(value: number) {
        this._departmentId = value;
    }

    get departmentName(): string {
        return this._departmentName;
    }

    set departmentName(value: string) {
        this._departmentName = value;
    }
}